source/CollisionSet.cpp

```diff
@@ -365,3 +369,10 @@ const vector<Body *> &CollisionSet::Ring(const Point &center, double inner, doub
 	}
 	return result;
 }
+
+
+
+const vector<Body *> &CollisionSet::All() const
```

Comment:
This CollisionSet.All() function looks quite elegant, but I'm afraid that the build-up of the "all" vector with all ships in the system for every frame (and also for non-system-wide hazards) is expensive (and might be better avoided).

Reply:
No more expensive than the creation of the rest of the collision set, right?